### PR TITLE
Add a mixer flag for store project and simplify other flags

### PR DESCRIPTION
**Pull request by shifucun**
This is to prepare for the stat var obs migration

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -77,8 +75,15 @@ func main() {
 		}
 	}
 
+	var branchCacheVersionBucket string
+	if *svobsMode {
+		branchCacheVersionBucket = "datcom-control"
+	} else {
+		branchCacheVersionBucket = "prophet_cache"
```
Comment (pradh) [inline cmd/main.go:82]: This should be "automation_control", no?

File: cmd/main.go [hunk 2]
```diff
@@ -37,25 +37,23 @@ import (
...
-	schemaPath       = flag.String("schema_path", "/translator/mapping", "The director that contains the schema mapping files")
+	mixerProject  = flag.String("mixer_project", "", "The cloud project to run the mixer instance.")
+	storeProject  = flag.String("store_project", "", "GCP project stores Bigtable and BigQuery.")
+	bqDataset     = flag.String("bq_dataset", "", "DataCommons BigQuery dataset.")
+	baseTableName = flag.String("base_table", "", "Base cache Bigtable table.")
+	port          = flag.Int("port", 12345, "Port on which to run the server.")
+	useALTS       = flag.Bool("use_alts", false, "Whether to use ALTS server authentication")
+	bigqueryOnly  = flag.Bool("bigquery_only", false, "The service only serves sparql query")
+	svobsMode     = flag.Bool("svobs_mode", false, "Use new storage that is built on top of StatVar instead of StatPop")
+	schemaPath    = flag.String("schema_path", "/translator/mapping", "The director that contains the schema mapping files")
```
Comment (pradh) [inline cmd/main.go:48]: Would we not have different mapping files based on svobsMode ?

Comment (pradh) [inline cmd/main.go:48]: nit: directory

File: cmd/main.go [hunk 3]
```diff
@@ -37,25 +37,23 @@ import (
...
-	svobsMode        = flag.Bool("svobs_mode", false, "Use new storage that is built on top of StatVar instead of StatPop")
-	schemaPath       = flag.String("schema_path", "/translator/mapping", "The director that contains the schema mapping files")
+	mixerProject  = flag.String("mixer_project", "", "The cloud project to run the mixer instance.")
+	storeProject  = flag.String("store_project", "", "GCP project stores Bigtable and BigQuery.")
+	bqDataset     = flag.String("bq_dataset", "", "DataCommons BigQuery dataset.")
+	baseTableName = flag.String("base_table", "", "Base cache Bigtable table.")
+	port          = flag.Int("port", 12345, "Port on which to run the server.")
+	useALTS       = flag.Bool("use_alts", false, "Whether to use ALTS server authentication")
+	bigqueryOnly  = flag.Bool("bigquery_only", false, "The service only serves sparql query")
+	svobsMode     = flag.Bool("svobs_mode", false, "Use new storage that is built on top of StatVar instead of StatPop")
```
Comment (pradh) [inline cmd/main.go:47]: nit: StatVar/StatVarObs instead of StatPop/Obs

File: docs/developer_guide.md [hunk 4]
```diff
@@ -96,14 +96,11 @@ Run the following code to start mixer gRPC server
 # cd into repo root directory
 
 go run cmd/main.go \
+    --mixer_project=datcom-mixer-staging \
+    --store_project=google.com:datcom-store-dev \
```
Comment (pradh) [inline docs/developer_guide.md:100]: nit: TODO: to switch this to datcom-store ?

File: cmd/main.go (hunk 1 above)
Comment (shifucun) [reply cmd/main.go:82]: That bucket only has the base cache version files: https://pantheon.corp.google.com/storage/browser/automation_control;tab=objects?forceOnBucketsSortingFiltering=false&project=google.com:datcom-store-dev&prefix=&forceOnObjectsSortingFiltering=false

File: docs/developer_guide.md (hunk 4 above)
Comment (shifucun) [reply docs/developer_guide.md:100]: Done

File: cmd/main.go (hunk 2 above)
Comment (shifucun) [reply cmd/main.go]: Done

Comment (shifucun) [reply cmd/main.go]: This is a mapped directory in the pod. It's the same for all deployment: https://github.com/datacommonsorg/mixer/blob/master/deploy/base/deployment.yaml#L64-L66, which is specified by the config map https://github.com/datacommonsorg/mixer/blob/master/deploy/base/deployment.yaml#L46-L47. For example, in the current deployment: https://github.com/datacommonsorg/mixer/blob/master/deploy/mapping/kustomization.yaml#L18

File: cmd/main.go (hunk 3 above)
Comment (shifucun) [reply cmd/main.go]: Done

**Review by shifucun — COMMENTED** [review]
thanks for review!

File: cmd/main.go (hunk 1 above)
Comment (pradh) [reply cmd/main.go:82]: Ah, I see!

**Review by pradh — APPROVED** [review]

**Pull request merged**